src/vcpkg/archives.cpp

```diff
@@ -177,6 +182,27 @@ namespace
 
 namespace vcpkg
 {
+    void extract_self_extracting_7z(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
+    {        
+        constexpr static const char header_7z[] =
+            {'7', 'z', 0xbc, 0xaf, 0x27, 0x1c};
+        
+        const auto subextension = Path(archive.stem()).extension();
```

Comment:
```suggestion
        const Path stem(archive.stem());
        const auto subextension = stem.extension();
```
I think subextension is of type `StringView`

Reply:
yeah and i am still surprised about this because std::filesystem normally returns a path.